x-pack/plugins/fleet/server/index.ts

```diff
@@ -27,7 +27,12 @@ export type {
   Artifact,
   ListArtifactsProps,
 } from './services';
-export { getRegistryUrl } from './services';
+export {
```

Comment:
fleet changes are purely for restricted path linting.

Reply:
Lets make sure that the Fleet team is comfortable with exposing these from the server side. They seem (to me) to be a bit too sensitive make available to others

Reply:
Perhaps a better approach would be to create a new service client and have that exposed to dependent plugins via the Plugin's `start()` or `setup()` contracts?

Reply:
Yes I feel exposing these methods is probably too sensitive as it's really internals of Fleet, and we should probably expose a client that allow to force reinstall one asset or something more generic from the start plugin

Reply:
In the ideal world I think we could have a package service exposed in the plugin start that we can get as the current user or as internal user (like the agents one) with a method `reeinstallAsset()`

```typescript
type FleetPluginStart = {
  packageClient: {
    asScoped(req: KibanaRequest): PackageClient
    asInternalUser: PackageClient
  }
}

type PackageClient = {
  reinstallAsset(pkgInfo: packageInfo, assetsPath: string[]): Promise<>
}
```

Reply:
Before we go forward with this, I'd really like to understand more about why the transform needs to be reinstalled. The contract between Fleet and Elasticsearch is supposed to be one where Fleet tells Elasticsearch what should exist and we shouldn't have to think about anything else.

I realize transforms are stateful so this is a bit different of an asset, but I really think we should be collaborating with the Elasticsearch team here on a better solution if this is the only one that currently exists. Where can I read up more on why this behavior is necessary and what problems are happening if we don't do this? There wasn't a linked issue or explanation of the motivation of this change.

Reply:
@joeypoon sent me some additional background offline and I understand more about the need for a stop gap solution here while some reliability issues in the transforms feature are worked on. +1 on @nchaulet's suggestion of providing a high-level API that takes asset paths and does the appropriate thing.

@joeypoon I'd like to see that service added as a separate PR from this one first and then used here before we merge this. Is this something you feel comfortable taking on? If not, we'll need to schedule this within the Fleet team's roadmap and prioritize accordingly.

Reply:
I'll give it a go 🙆‍♂️ .